### PR TITLE
Finalize Feature/iat 523

### PR DESCRIPTION
**Pull request by alex-ponce**
**Include the selected item tab in URL**
* Formatted routing configuration
* Added tabEvent event emitter to notify crud component of the current selected tab
* Changed loadItem() signature to allow selected tab to be passed in

1. This feature now allows links to items and the tab to display by using any of these three URL variations:
```
/item/MC34/history
/item/MC34/stimulus
/item/MC34/workflow
```
If no tab identifier is passed 'history' is used as default

2. When editing an item, the tab that was selected prior to clicking the Edit button will now be displayed

### CODE REVIEW

File: src/app/item/crud/item-crud.component.html [hunk 1]
```diff
@@ -89,6 +89,7 @@
                    [itemType]="item.itemType"
                    [selected]="selectedTab"
                    [isReadOnly]="isBeingViewedByCurrentUser"
+                   (tabChanged)="onTabChanged($event)"
```
Comment (bdrainer) [inline src/app/item/crud/item-crud.component.html]: nice

File: src/app/item/crud/item-crud.component.ts [hunk 2]
```diff
@@ -72,7 +69,12 @@ export class ItemCrudComponent implements OnInit {
   }
 
   ngOnInit() {
-    this.loadItem();
+    this.route.params
+      .subscribe(
+        params => {
+          // Extract item ID and selected Tab from route
+          this.loadItem(params['id'], params['tab']);
```
Comment (bdrainer) [inline src/app/item/crud/item-crud.component.ts]: makes sense

File: src/app/item/crud/item-crud.component.ts [hunk 3]
```diff
@@ -110,43 +112,41 @@ export class ItemCrudComponent implements OnInit {
       .beginEditTransaction(this.item.id, "Began edit")
       .subscribe(
         () => {
-          this.loadItem();
+          this.loadItem(this.item.id, this.selectedTab);
```
Comment (bdrainer) [inline src/app/item/crud/item-crud.component.ts]: this is how you are using the component values which are likely changing for the selected tab as the user changes it

File: src/app/item/crud/tabs/item-tabs.component.ts [hunk 4]
```diff
@@ -44,6 +45,7 @@ export class ItemTabsComponent {
     this.selected = tab;
     // Update URL to reflect new tab. This does NOT cause the entire page to reload
     this.location.go("/item/" + this.item.id + "/" + tab);
+    this.tabChanged.emit(this.selected);
```
Comment (bdrainer) [inline src/app/item/crud/tabs/item-tabs.component.ts]: Is there a place where the item crud component registers to listen for this change?

Comment (alex-ponce) [inline src/app/item/crud/tabs/item-tabs.component.ts]: Actually, this is where the item-tabs component "broadcasts" the new selected tab name.

Here's where we configure in the crud component what to do with this emitted value:

https://github.com/SmarterApp/AP_ItemAuthoringTool/blob/develop/src/app/item/crud/item-crud.component.html#L92

**Pull request merged**